sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -1161,6 +1162,7 @@ impl ty::TyExpression {
             (false, Some(func_decl), None, None) => {
                 warnings.append(&mut function_probe_warnings);
                 errors.append(&mut function_probe_errors);
+                let args = args.unwrap_or_default();
```

Comment:
Is there any case in which this assumption would be incorrect? i.e. is there any code that the user could write that would trigger this to fail silently?

Reply:
It was possible to have args set to None and we could have functions called with no parenthesis. Thanks for noticing. Recent changes address this issue, and add a test case as well.